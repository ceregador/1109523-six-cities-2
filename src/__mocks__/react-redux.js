export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
