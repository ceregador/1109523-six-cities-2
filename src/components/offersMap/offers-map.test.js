import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import Renderer from 'react-test-renderer';
import OffersMap from '../offersMap/offers-map.jsx';

jest.mock(`../rentObjectCardList/rent-object-card-list.jsx`);
jest.mock(`react-redux`);
jest.mock(`../../selectors/selector`);

it(`renders and connects correctly`, () => {
  const tree = Renderer
    .create(
        <OffersMap
          cityCoordinates={[]}
          offersCoordinates={[]}
        />)
    .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), null);

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({cityOffers: []});
  expect(mappedProps).toHaveProperty(`offersCoordinates`);

  expect(connectAdvanced).toHaveBeenCalledWith(OffersMap);

  expect(tree).toMatchSnapshot();
});
