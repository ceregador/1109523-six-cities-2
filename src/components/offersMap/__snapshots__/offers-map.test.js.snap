// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and connects correctly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "800px",
    }
  }
/>
`;
